drivers/scd30/scd30.c

```diff
@@ -0,0 +1,396 @@
+    if (ret != 0) {
+        i2c_release(SCD30_I2C);
+        return ret;
+    }
+    i2c_release(SCD30_I2C);
+
+    xtimer_usleep(SCD30_READ_WRITE_SLEEP_US);
+
+    i2c_acquire(SCD30_I2C);
+    ret = i2c_read_bytes(SCD30_I2C, SCD30_I2C_ADDRESS, buffer, 3, 0);
+    i2c_release(SCD30_I2C);
+
+    if (ret != 0) {
+        DEBUG("[scd30] Problem reading param %u from sensor\n", param);
+    }
+
+    if (_scd30_crc_check(buffer, 2, buffer[2]) != 0) {
+        return SCD30_CRC_ERROR;
+    }
+
+    *val = (buffer[0] << 8) | buffer[1];
+    return SCD30_OK;
+}
+
+int8_t scd30_read_triggered(scd30_t *dev, scd30_measurement_t *result,
+                            uint16_t *apc)
+{
+    uint16_t state = 0;
+    uint16_t curr_interval;
+    uint32_t initial_time;
+    int ret;
+
+    if (!_is_valid_press_comp(apc)) {
+        DEBUG("[scd30] Ambient pressure compensation value out of range\n");
+        return SCD30_INVALID_VALUE;
+    }
+
+    ret = scd30_get_param(dev, SCD30_INTERVAL, &curr_interval);
+    if (ret != 0) {
+        return ret;
+    }
+    if (curr_interval != SCD30_MIN_INTERVAL) {
+        ret = scd30_set_param(dev, SCD30_INTERVAL, SCD30_MIN_INTERVAL);
+        if (ret != 0) {
+            DEBUG("[scd30] Error setting interval to min\n");
+            return ret;
+        }
+    }
+
+    /* Doing this to reduce the traffic on the I2C bus as
+     * compared when constantly polling the device for status
+     */
+    xtimer_sleep(SCD30_MIN_INTERVAL);
+
+    initial_time = xtimer_now_usec();
+    ret = scd30_get_param(dev, SCD30_STATUS, &state);
+    if (ret != 0) {
+        return ret;
+    }
+    // DEBUG("[scd30] state :%d\n", state);
+    while (state != 1 &&
+           (xtimer_now_usec() - initial_time) < SCD30_DATA_RDY_TIMEOUT) {
+        // while (state != 1) {
+        ret = scd30_get_param(dev, SCD30_STATUS, &state);
+        if (ret != 0) {
+            return ret;
+        }
+        // DEBUG("[scd30] state :%d\n", state);
+    }
+    gpio_toggle(test_pin);
+    ret = _scd30_read_data(dev, result);
+    gpio_toggle(test_pin);
+    if (ret != SCD30_OK) {
+        DEBUG("[scd30] Error reading data \n");
+        return ret;
+    }
+
+    if (curr_interval != SCD30_MIN_INTERVAL) {
+        ret = scd30_set_param(dev, SCD30_INTERVAL, curr_interval);
+        if (ret != 0) {
+            DEBUG("[scd30] Error resetting interval to original value\n");
+            return ret;
+        }
+    }
+    return SCD30_OK;
+}
+
+uint8_t scd30_read_periodic(scd30_t *dev, scd30_measurement_t *result)
+{
+    uint16_t state = 0;
+    uint32_t initial_time;
+    int ret = 0;
+
+    initial_time = xtimer_now_usec();
+    ret = scd30_get_param(dev, SCD30_STATUS, &state);
+    if (ret != 0) {
+        return ret;
+    }
+
+    while (state != 1 &&
+           (xtimer_now_usec() - initial_time) < SCD30_DATA_RDY_TIMEOUT) {
+        scd30_get_param(dev, SCD30_STATUS, &state);
+    }
+    if (state == 1) {
```

Comment:
You can reduce the level of nesting required.

```suggestion
    if (state != 1) {
        return SCD30_NO_NEW_DATA;
    }
```